src/components/keyboardnavigation.js

```diff
@@ -126,6 +126,41 @@ define(["inputManager", "layoutManager"], function (inputManager, layoutManager)
         });
     }
 
+    function isGamepadConnected() {
+        var gamepads = navigator.getGamepads();
+        var i;
+        var len;
+        for (i = 0, len = gamepads.length; i < len; i++) {
+            var gamepad = gamepads[i];
+            if (gamepad) {
+                console.log(gamepad);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    function attachGamepad(e) {
+        if (isGamepadConnected()) {
+            require(["components/input/gamepadtokey"]);
+            console.log("Gamepad connected! Attaching gamepadtokey.js script");
+        }
+    }
+
+    function dettachGamepad(e) {
+        if (!isGamepadConnected()) {
+            delete require.cache[require(["components/input/gamepadtokey"])];
```

Comment:
`require.cache` is undefined - I got an error, and `gamepadtokey` stays and loops through 4 virtual gamepads endlessly.

Reply:
This is where the problem you describe above happens. Whenever a gamepad is disconnected, the ``gamepadtokey.js`` should be killed no matter what, so the "check" for connected gamepads in ``gamepadtokey.js`` rely solely in ``keyboardnavigation.js``, which is the responsible for killing the script whenever no gamepads are connected.

This is the most common approach I found for "unrequiring" a previously required module and kill the execution of ``gamepadtokey.js``. Couldn't test it as I have no gamepads, so thank you for pointing that out, will need to investigate that more.

Do you know what's the best way and how I can "unrequire" a module?

Reply:
I am absolute dumb in "requirejs" :neutral_face: Like you, I found this approach to remove the "required" module. I may be wrong, but problem is that `require` is provided by `amelada` library, so it may not be fully implemented.

As I mention before, if we have no option to remove "required" module, we can start/stop `runInputLoop`: `requestAnimationFrame`/`cancelAnimationFrame`. In this case, we can (or have to) move all gamepad-related code into `gamepadtokey` and just require it in `site.js` (near `keyboardnavigation`).